### DISCUSSION TITLE
How Streams provided by watch() are managed?

### DISCUSSION THREAD

**Discussion by givip** [simolus3/drift#1024] [Q&A] [answered]
If I simply cancel subscription is it enough to right memory management?
If I call watch multiple times is it returns the same stream or the new one?

**Comment by simolus3** [comment, marked as answer]
> If I simply cancel subscription is it enough to right memory management?

Yes, moor will evict cached data shortly after the last listener detaches from a stream.

>  If I call watch multiple times is it returns the same stream or the new one?

If you call `watch()` while another listener is active, the stream will be re-used. You might get a different stream instance, but they will be powered by the same underlying `StreamController`. Further, moor will cache the latest value of a stream so that the query won't be run if it's not necessary.
